google-cloud-bigquery-storage/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/storage/javadsl/GoogleBigQueryStorage.scala

```diff
@@ -0,0 +1,169 @@
+           readOptions: TableReadOptions): Source[Source[GenericRecord, NotUsed], CompletionStage[NotUsed]] =
+    read(projectId, datasetId, tableId, Some(readOptions), 0)
+
+  /**
+   * Create a source that contains a number of sources, one for each stream, or section of the table data.
+   * These sources will emit one GenericRecord for each row within that stream.
+   *
+   * @param projectId   the projectId the table is located in
+   * @param datasetId   the datasetId the table is located in
+   * @param tableId     the table to query
+   * @param readOptions TableReadOptions to reduce the amount of data to return, either by column projection or filtering
+   * @param maxNumStreams An optional max initial number of streams. If unset or zero, the server will provide a value of streams so as to produce reasonable throughput.
+   *                      Must be non-negative. The number of streams may be lower than the requested number, depending on the amount parallelism that is reasonable for the table.
+   *                      Error will be returned if the max count is greater than the current system max limit of 1,000.
+   */
+  def read(projectId: String,
+           datasetId: String,
+           tableId: String,
+           readOptions: TableReadOptions,
+           maxNumStreams: Int): Source[Source[GenericRecord, NotUsed], CompletionStage[NotUsed]] =
+    read(projectId, datasetId, tableId, Some(readOptions), maxNumStreams)
+
+  /**
+   * Create a source that contains a number of sources, one for each stream, or section of the table data.
+   * These sources will emit one GenericRecord for each row within that stream.
+   *
+   * @param projectId   the projectId the table is located in
+   * @param datasetId   the datasetId the table is located in
+   * @param tableId     the table to query
+   * @param maxNumStreams An optional max initial number of streams. If unset or zero, the server will provide a value of streams so as to produce reasonable throughput.
+   *                      Must be non-negative. The number of streams may be lower than the requested number, depending on the amount parallelism that is reasonable for the table.
+   *                      Error will be returned if the max count is greater than the current system max limit of 1,000.
+   */
+  def read(projectId: String,
+           datasetId: String,
+           tableId: String,
+           maxNumStreams: Int): Source[Source[GenericRecord, NotUsed], CompletionStage[NotUsed]] =
+    read(projectId, datasetId, tableId, None, maxNumStreams)
+
+  private val RequestParamsHeader = "x-goog-request-params"
+
+  private[this] def read(
+      projectId: String,
+      datasetId: String,
+      tableId: String,
+      readOptions: Option[TableReadOptions],
+      maxNumStreams: Int
+  ): Source[Source[GenericRecord, NotUsed], CompletionStage[NotUsed]] =
+    Source.setup[Source[GenericRecord, NotUsed], NotUsed] { (mat: ActorMaterializer, attr: Attributes) =>
```

Comment:
What is the reason for a duplication of the implementation between scaladsl and javadsl?
In most cases, it should be possible to use the scaladsl implementation and amend it with conversions to they Java counterparts.